includes/AMP/Canonical_Sanitizer.php

```diff
@@ -54,28 +54,43 @@ class Canonical_Sanitizer extends AMP_Base_Sanitizer {
 	 * @return void
 	 */
 	public function sanitize() {
-		$links         = [];
+		/**
+		 * The rel=canonical DOM element.
+		 *
+		 * @var DOMElement The element.
+		 */
+		$rel_canonical = null;
 		$link_elements = $this->dom->head->getElementsByTagName( Tag::LINK );
 
+		// This fallback to get_permalink() ensures that there's a canonical link
+		// even when previewing drafts.
+		$canonical_url = wp_get_canonical_url();
+		if ( ! $canonical_url ) {
+			$canonical_url = get_permalink();
+		}
```

Comment:
If are you viewing as a preview, should this value be `get_preview_post_link`?

Reply:
If you just visit a draft directly, e.g. `example.com/?p=12345`, then this wouldn't apply. `get_preview_post_link()` is just `get_permalink()` + `?preview=true` anyway.

So `get_permalink()` is fine IMO.